perfkitbenchmarker/scripts/object_storage_api_tests.py

```diff
@@ -271,81 +262,238 @@ def PercentileCalculator(numbers):
   return result
 
 
-def _ListObjects(storage_schema, bucket, prefix, host_to_connect=None):
-  """List objects under a bucket given a prefix.
+# ### Backends ###
 
-  Args:
-    storage_schema: The address schema identifying a storage. e.g., "gs"
-    bucket: Name of the bucket.
-    prefix: A prefix to list from.
-    host_to_connect: An optional endpoint string to connect to.
+class ObjectStorageBackend(object):
```

Comment:
I'd put this script and the provider specific scripts into one single directory here, then the high level driver script only needs to copy everything under that directory to the target VM, that's it. 

I'm fine with doing it in a separate CR, but let's make sure we put a TODO here, and also file an issue to track this.